firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionBlockedTrackersView.swift

```diff
@@ -0,0 +1,162 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import Common
+
+final class TrackingProtectionBlockedTrackersView: UIView, ThemeApplicable {
+    private struct UX {
+        static let trackersLabelConstraintConstant = 16.0
+    }
+
+    var trackersButtonCallback: (() -> Void)?
+
+    private let shieldImage: UIImageView = .build { image in
+        image.contentMode = .scaleAspectFit
+        image.clipsToBounds = true
+        image.layer.masksToBounds = true
+        image.image = UIImage(imageLiteralResourceName: StandardImageIdentifiers.Large.shield)
+            .withRenderingMode(.alwaysTemplate)
+    }
+
+    private let trackersLabel: UILabel = .build { label in
+        label.font = FXFontStyles.Regular.body.scaledFont()
+        label.numberOfLines = 0
+        label.adjustsFontForContentSizeCategory = true
+    }
+
+    private let trackersDetailArrow: UIImageView = .build { image in
+        image.image = UIImage(imageLiteralResourceName: StandardImageIdentifiers.Large.chevronLeft)
+            .withRenderingMode(.alwaysTemplate)
+            .imageFlippedForRightToLeftLayoutDirection()
+        image.transform = CGAffineTransform(rotationAngle: .pi)
+    }
+
+    private let trackersHorizontalLine: UIView = .build()
+    private let trackersButton: UIButton = .build()
+
+    private var shieldImageHeightConstraint: NSLayoutConstraint?
+    private var trackersArrowHeightConstraint: NSLayoutConstraint?
+    private var trackersLabelTopConstraint: NSLayoutConstraint?
+    private var trackersLabelBottomConstraint: NSLayoutConstraint?
+
+    init() {
+        super.init(frame: .zero)
+        setupLayout()
+    }
+
+    required init(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    private func setupLayout() {
+        self.addSubviews(shieldImage, trackersLabel, trackersDetailArrow, trackersButton, trackersHorizontalLine)
+        shieldImageHeightConstraint = shieldImage.heightAnchor.constraint(equalToConstant: TPMenuUX.UX.iconSize)
+        trackersArrowHeightConstraint = trackersDetailArrow.heightAnchor.constraint(
+            equalToConstant: TPMenuUX.UX.iconSize
+        )
+
+        trackersLabelTopConstraint = trackersLabel.topAnchor.constraint(
+            equalTo: self.topAnchor,
+            constant: UX.trackersLabelConstraintConstant)
+        trackersLabelBottomConstraint = trackersLabel.bottomAnchor.constraint(
+            equalTo: self.bottomAnchor,
+            constant: -UX.trackersLabelConstraintConstant)
+
+        NSLayoutConstraint.activate([
+            shieldImage.leadingAnchor.constraint(
+                equalTo: self.leadingAnchor,
+                constant: TPMenuUX.UX.horizontalMargin
```

Comment:
I don't have a lot of context for this task... but would it make sense to split out the `UX` constants into each custom view as well?

Same goes for anywhere else there are `TPMenuUX` constants.

Reply:
horizontalMargin is used in many files. Let's keep it as it is instead of duplicate it in many files.